### PR TITLE
Pin Registration SC-3635

### PR DESCRIPTION
**Pull request by tofixx**
# Checkliste
See https://ticketsystem.schul-cloud.org/browse/SC-3635 for what this does
https://github.com/schul-cloud/schulcloud-server/pull/1062

### CODE REVIEW

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

File: controllers/registration.js [hunk 1]
```diff
@@ -105,7 +105,15 @@ ${res.locals.theme.short_title}-Team`,
 			res.sendStatus(200);
 		})
 		.catch((err) => {
-			res.status(500).send((err.error || {}).message || err.message || 'Fehler bei der Registrierung.');
+			let message = 'Ein unbekannter Fehler ist aufgetreten. Bitte versuche es erneut.'
```
Comment (dominikbruechner) [inline controllers/registration.js]:
```suggestion
			let message = 'Hoppla, ein unbekannter Fehler ist aufgetreten. Bitte versuche es erneut.'
```

File: static/scripts/dataprivacy/pin.js [hunk 2]
```diff
@@ -14,31 +17,33 @@ function sendPin(sendConfirm) {
 		data: { email: usermail, mailTextForRole: role },
 	}).done(() => {
 		if (sendConfirm) {
-			$.showNotification(`Eine PIN wurde erfolgreich an ${usermail} versendet.`, 'success', 15000);
+			$.showNotification(`Eine PIN wurde an ${usermail} versendet.`, 'success', 15000);
```
Comment (dominikbruechner) [inline static/scripts/dataprivacy/pin.js]:
```suggestion
			$.showNotification(`Wir haben dir soeben einen Bestätigungscode an ${usermail} gesendet.`, 'success', 15000);
```

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

File: static/scripts/dataprivacy/pin.js [hunk 3]
```diff
@@ -14,31 +17,33 @@ function sendPin(sendConfirm) {
...
 		}
 		pinSent = true;
 	}).fail(() => {
 		const errorMessage = `Fehler bei der PIN-Erstellung!
-			Bitte versuche es mit 'Code erneut zusenden' und prüfe deine E-Mail-Adresse (${usermail}).`;
+			Bitte versuche es erneut und prüfe deine E-Mail-Adresse (${usermail}).`;
```
Comment (dominikbruechner) [inline static/scripts/dataprivacy/pin.js]:
```suggestion
			Bitte prüfe deine E-Mail-Adresse (${usermail}) und versuche es dann erneut.`;
```

File: static/scripts/dataprivacy/pin.js [hunk 4]
```diff
@@ -14,31 +17,33 @@ function sendPin(sendConfirm) {
...
-	$('.pin-input .digit').val('');
+	if (timeoutSend === false) {
+		sendPin(true);
+		$('.pin-input .digit').val('');
+		timeoutSend = true;
+		resendTimer = setTimeout(() => {
+			timeoutSend = false;
+		}, 60000);
+	} else {
+		$.showNotification(`Eine E-Mail wurde bereits versendet. Bitte prüfe, ob du wirklich keine E-Mail erhalten hast. Nach einer Minute kannst du die E-Mail erneut anfordern.`, 'info', 7000);
```
Comment (dominikbruechner) [inline static/scripts/dataprivacy/pin.js]:
```suggestion
		$.showNotification(`Wir haben dir bereits eine E-Mail gesendet. Bitte prüfe auch deinen Spam-Ordner, ob du wirklich keine E-Mail erhalten hast. Nach einer Minute kannst du den Code erneut anfordern.`, 'info', 7000);
```

File: static/scripts/dataprivacy/pin.js [hunk 5]
```diff
@@ -14,31 +17,33 @@ function sendPin(sendConfirm) {
 		data: { email: usermail, mailTextForRole: role },
 	}).done(() => {
 		if (sendConfirm) {
-			$.showNotification(`Eine PIN wurde erfolgreich an ${usermail} versendet.`, 'success', 15000);
+			$.showNotification(`Eine PIN wurde an ${usermail} versendet.`, 'success', 15000);
 		}
 		pinSent = true;
 	}).fail(() => {
 		const errorMessage = `Fehler bei der PIN-Erstellung!
```
Comment (dominikbruechner) [inline static/scripts/dataprivacy/pin.js]:
```suggestion
		const errorMessage = `Hoppla, es gab einen Fehler bei der Code-Erstellung!
```

File: views/registration/pin.hbs [hunk 6]
```diff
@@ -1,10 +1,10 @@
 <div class="pin-input {{class}}">
-  <input class="combined-pin" type="text" name="{{name}}" pattern="{{pattern}}{ {{~digits~}} }" {{#if required}}required{{/if}} tabIndex="-1"/>
+  Bestätigungscode: <input class="combined-pin" type="text" name="{{name}}" pattern="{{pattern}}{ {{~digits~}} }" {{#if required}}required{{/if}} tabIndex="-1"/>
   {{#times digits}}
     <input class="digit" type="text" pattern="{{../pattern~}} {1}" maxlength="1" placeholder="·" {{#if ../required}}required{{/if}}>
   {{/times}}
 </div>
-<p class="pin-invalidated hidden">Die zuletzt übermittelte PIN ist nun verbraucht. Wenn etwas nicht funktioniert hat, generiere dir bitte mittels "Code erneut zusenden" eine neue PIN und gib diese hier ein.</p>
+<p class="pin-invalidated hidden">Die zuletzt übermittelte PIN ist nun verbraucht. Wenn etwas nicht funktioniert hat, generiere dir bitte mittels "Code jetzt anfordern" eine neue PIN und gib diese hier ein.</p>
```
Comment (dominikbruechner) [inline views/registration/pin.hbs]:
```suggestion
<p class="pin-invalidated hidden">Der zuletzt übermittelte Code ist nun verbraucht. Wenn etwas nicht funktioniert hat, generiere dir bitte mittels "Code jetzt anfordern" einen neuen Code und gib diesen hier ein.</p>
```

File: views/registration/request-pin.hbs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<p>Bitte bestätige uns, die Korrektheit der angegebenen E-Mail-Adresse. <br>Fordere dazu einen Bestätigungscode an:</p>
```
Comment (dominikbruechner) [inline views/registration/request-pin.hbs]:
```suggestion
<p>Wir müssen nur kurz prüfen, ob es deine E-Mail-Adresse wirklich gibt. <br>Bitte fordere dazu einen Bestätigungscode an:</p>
```

File: views/registration/request-pin.hbs [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+<p><button type="button" class="btn btn-primary" id="resend-pin">Code jetzt anfordern</button></p>
```
Comment (dominikbruechner) [inline views/registration/request-pin.hbs]:
```suggestion
<p><button type="button" class="btn btn-primary" id="resend-pin">Bestätigungscode jetzt anfordern</button></p>
```

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5201181&bid=17061114)

**Pull request merged**